### DISCUSSION TITLE
Why does VS Code Pylance not recognise PyMuPDF methods?

### DISCUSSION THREAD

**Discussion by paultheman** [pymupdf/PyMuPDF#2545] [Q&A] [answered]
I was wondering why there is no code completion for some of the methods of the PyMuPDF package. For example fitz.Document is recognised but fitz.open is not.

![image](https://github.com/pymupdf/PyMuPDF/assets/38941045/45bfacb3-cee7-4ced-9f23-d8293c7929cb)

Thank you!

**Comment by JorjMcKie** [comment, marked as answer]
This method and several others are defined in `__init__.py` and depending on the IDE this sometimes is beyond its wits. Works under MS Code for example or even in Python IDLE.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
From: paultheman ***@***.***>
Sent: Saturday, July 15, 2023 4:22:54 PM
To: pymupdf/PyMuPDF ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [pymupdf/PyMuPDF] Why does VS Code Pylance not recognise PyMuPDF methods? (Discussion #2545)


I was wondering why there is no code completion for some of the methods of the PyMuPDF package. For example fitz.Document is recognised but fitz.open is not.

[image]<https://user-images.githubusercontent.com/38941045/253769434-45bfacb3-cee7-4ced-9f23-d8293c7929cb.png>

Thank you!
